SevenSlots/SevenSlots/SevenSlots/View/SlotMachineView.xaml.cs

```diff
@@ -0,0 +1,177 @@
+using System.Threading.Tasks;
+
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+using SevenSlots.Helpers;
+
+namespace SevenSlots.View
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class SlotMachineView : ContentPage
+    {
+        public SlotMachineView()
+        {
+            InitializeComponent();
+        }
+
+        protected override void OnAppearing()
+        {
+            base.OnAppearing();
+            MessagingCenter.Send(this, "AllowLandscape");
+        }
+
+        //during page close setting back to portrait
+        protected override void OnDisappearing()
+        {
+            base.OnDisappearing();
+            MessagingCenter.Send(this, "PreventLandscape");
+            (BindingContext as SlotMachineViewModel).UpdateWallet();
+
+            //Save the wallet locally as well
+            string userString = JsonSerializer.Serialize((BindingContext as SlotMachineViewModel).User);
+            Session.GeneralSettings = userString;
+        }
+        private string numberToImageSource(int nr)
+        {
+            if(nr < 3)
+            {
+                return "CherrySlot.png";
+            }
+            else if(nr < 6)
+            {
+                return "GrapeSlot.png";
+            }else if(nr < 9)
+            {
+                return "LemonSlot.png";
+            }else if (nr < 12) 
+            {
+                return "OrangeSlot.png";
+            }else if(nr < 14)
+            {
+                return "SevenSlot.png";
+            }
+            return "SpecialSlot.png";
+        }
+
+        private double getWin(string img1, string img2, string img3)
+        {
+            double bet = (BindingContext as SlotMachineViewModel).Bet;
+
+            if (img1 == "File: SpecialSlot.png" && img1 == img2 && img1 == img3)
+            {
+                return bet * 150;
+            }
+
+            if (img1 == "File: SevenSlot.png" && img1 == img2 && img1 == img3)
+            {
+                return bet * 60;
+            }
+
+            if((img1 == "File: SevenSlot.png" || img1 == "File: SpecialSlot.png") &&
+               (img2 == "File: SevenSlot.png" || img2 == "File: SpecialSlot.png") ||
+               (img1 == "File: SevenSlot.png" || img1 == "File: SpecialSlot.png") &&
+               (img3 == "File: SevenSlot.png" || img3 == "File: SpecialSlot.png") ||
+               (img2 == "File: SevenSlot.png" || img2 == "File: SpecialSlot.png") &&
+               (img3 == "File: SevenSlot.png" || img3 == "File: SpecialSlot.png"))
+            {
+                return bet * 30;
+            }
+
+            if(img1 == img2 && (img3 == "File: SevenSlot.png" || img3 == "File: SpecialSlot.png") ||
+                img1 == img3 && (img2 == "File: SevenSlot.png" || img2 == "File: SpecialSlot.png") ||
+                img2 == img3 && (img1 == "File: SevenSlot.png" || img1 == "File: SpecialSlot.png"))
+            {
+                return bet * 30;
+            }
+
```

Comment:
I suggest to try some similar approach with a dictionary, although I guess this would be harder, so if you can't find an easy way you can leave it as it is